### PR TITLE
[Fix]: cls pretrained mode and tune space

### LINKED ISSUE #2169: Hyperparameter Tuning with Ray Tune
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

_No response_

### Bug

**fl_gamma** should be removed in default search space description when we do **Hyperparameter Tuning with Ray Tune**.

**_fl_gamma | tune.uniform(0.0, 2.0) | Focal loss gamma_**

### Environment

_No response_

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### LINKED ISSUE #2196: Classify training, pretrained weights are not correctly transferred as expected
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

The argument "pretrained" is forced to be "False".  So any passed pretrained model weights will be reset to default parameters.

https://github.com/ultralytics/ultralytics/blob/d0b616e41e827e1ac65ee99da1339e707a984989/ultralytics/yolo/v8/classify/train.py#L31-L34

The following code may solve the problem.
` pretrained = self.args.pretrained`

### Environment

_No response_

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by AyushExel**
Fixes: https://github.com/ultralytics/ultralytics/issues/2169 ,  https://github.com/ultralytics/ultralytics/issues/2196

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Ultralytics has refined hyperparameter tuning and model initialization options.

### 📊 Key Changes
- Removed the search space parameter `fl_gamma` for Focal Loss gamma from hyperparameter tuning.
- Updated the model training code to conditionally load pretrained weights based on a provided argument.

### 🎯 Purpose & Impact
- The removal of `fl_gamma` in hyperparameter tuning could mean the team found it to be non-essential or plans to manage it differently, potentially simplifying the tuning process.
- Allowing the option to load pretrained weights more flexibly enables users to choose whether they start with a pretrained model, which can lead to better initial performance and faster convergence in training new models.

### CODE REVIEW

**Comment by Laughing-q** [comment]
@AyushExel I don't really know the logic of classification but shouldn't `pretrained` be `True` when `weights is not None`? since `model` has already loaded `weights`.

**Comment by AyushExel** [comment]
Yeah me neither that's why I didn't want to touch it 🤣 
But I think the usage is like this - `pretrained` controls whether or not the model is reset. So, even if `weights` is provided, setting `pretrained=Fasle` will reset it. It's something the user should set and not to be inferred by us.
